plasma/root_chain/contracts/RootChain/RootChain.sol

```diff
@@ -123,32 +90,34 @@ contract RootChain {
 
     // @dev Allows anyone to deposit funds into the Plasma chain
     // @param txBytes The format of the transaction that'll become the deposit
-    // TODO: This needs to be optimized so that the transaction is created
-    //       from msg.sender and msg.value
-    function deposit(bytes txBytes)
+    function deposit()
         public
         payable
     {
         require(currentDepositBlock < childBlockInterval);
-        var txList = txBytes.toRLPItem().toList(11);
-        require(txList.length == 11);
-        for (uint256 i; i < 6; i++) {
-            require(txList[i].toUint() == 0);
-        }
-        require(txList[7].toUint() == msg.value);
-        require(txList[9].toUint() == 0);
         bytes32 zeroBytes;
-        bytes32 root = keccak256(keccak256(txBytes), new bytes(130));
-        for (i = 0; i < 16; i++) {
+        bytes32 root = keccak256(msg.sender, msg.value);
+        for (uint i = 0; i < 16; i++) {
             root = keccak256(root, zeroBytes);
             zeroBytes = keccak256(zeroBytes, zeroBytes);
         }
-        childChain[getDepositBlock()] = childBlock({
+        uint256 depositBlock = getDepositBlock();
+        childChain[depositBlock] = childBlock({
             root: root,
             created_at: block.timestamp
         });
         currentDepositBlock = currentDepositBlock.add(1);
-        Deposit(txList[6].toAddress(), txList[7].toUint());
+        Deposit(msg.sender, msg.value, depositBlock);
+    }
+
+    function startDepositExit(uint256 utxoPos, uint256 amount, bytes proof)
```

Comment:
root can be computed the same way as in deposit function, so it doesn't seem necessary to send a proof to exit a deposit, right?

Reply:
@dshulyak You're right, I'll take that out.  Thanks!